airflow/models.py

```diff
@@ -346,10 +346,10 @@ def kill_zombies(self, session=None):
 
                     # now set non db backed vars on ti
                     ti.task = task
-                    ti.test_mode = False
+                    ti.test_mode = configuration.getboolean('core', 'unit_test_mode')
```

Comment:
@bolkedebruin as per my understanding, if a task ran in `test_mode` it would not have db entries... and as `kill_zombie` picks up `TIs` via a db query, its guaranteed that all entries it picked are not of `test_mode`...